### PR TITLE
doc: move condition from where clause to on condition in nt-dml

### PR DESCRIPTION
**Pull request by ekexium**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add a frequently used tip for nt-dml to the FAQ section.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.4 (TiDB 7.4 versions)
- [x] v7.3 (TiDB 7.3 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: non-transactional-dml.md [hunk 1]
```diff
@@ -327,6 +327,33 @@ batch-dml 是一种在 DML 语句执行期间将一个事务拆成多个事务
 
 ## 常见问题
 
+### 在多表 join 语句中，"Unknown column xxxx in 'where clause'" 错误
```
Comment (qiancai) [inline non-transactional-dml.md:330]:
```suggestion
### 执行多表 join 语句后，遇到 `Unknown column xxxx in 'where clause` 错误
```

File: non-transactional-dml.md [hunk 2]
```diff
@@ -327,6 +327,33 @@ batch-dml 是一种在 DML 语句执行期间将一个事务拆成多个事务
...
+
+这经常是拼接查询语句时，只查询划分列所在的表，但是 where 子句会直接拼接在这个查询语句中。如果 where 子句中的条件涉及到了其它表，就会出现这个错误。例如：
+
+```sql
+batch on test.t2.id limit 1 insert into t select t2.id, t2.v, t3. from t2, t3 where t2.id = t3.id
```
Comment (qiancai) [inline non-transactional-dml.md:335]:
```suggestion
batch on test.t2.id LIMIT 1 INSERT INTO t SELECT t2.id, t2.v, t3. FROM t2, t3 WHERE t2.id = t3.id
```

File: non-transactional-dml.md [hunk 3]
```diff
@@ -327,6 +327,33 @@ batch-dml 是一种在 DML 语句执行期间将一个事务拆成多个事务
 
 ## 常见问题
 
+### 在多表 join 语句中，"Unknown column xxxx in 'where clause'" 错误
+
+这经常是拼接查询语句时，只查询划分列所在的表，但是 where 子句会直接拼接在这个查询语句中。如果 where 子句中的条件涉及到了其它表，就会出现这个错误。例如：
```
Comment (qiancai) [inline non-transactional-dml.md:332]:
```suggestion
在拼接查询语句时，通常只会考虑所需拼接的列所在的表，但是 `WHERE` 子句会直接拼接在这个查询语句中。如果 `WHERE` 子句中的条件涉及到了其他表，就会出现该错误。例如：
```

File: non-transactional-dml.md [hunk 4]
```diff
@@ -327,6 +327,33 @@ batch-dml 是一种在 DML 语句执行期间将一个事务拆成多个事务
...
+
+```sql
+(1054, "Unknown column '3.id' in 'where clause'")
+```
+可以通过 dry run query 打印出查询语句来确认。
+
+这种情况下可以尝试将 where 子句中涉及其它表的条件移动到 join 的 on 条件中，例如：
+
+```sql
+batch on test.t2.id limit 1 insert into t select t2.id, t2.v, t3. from t2 join t3 on t2.id=t3.id
```
Comment (qiancai) [inline non-transactional-dml.md:346]:
```suggestion
batch on test.t2.id LIMIT 1 INSERT INTO t SELECT t2.id, t2.v, t3. FROM t2 JOIN t3 ON t2.id=t3.id
```

File: non-transactional-dml.md [hunk 5]
```diff
@@ -327,6 +327,33 @@ batch-dml 是一种在 DML 语句执行期间将一个事务拆成多个事务
...
+这经常是拼接查询语句时，只查询划分列所在的表，但是 where 子句会直接拼接在这个查询语句中。如果 where 子句中的条件涉及到了其它表，就会出现这个错误。例如：
+
+```sql
+batch on test.t2.id limit 1 insert into t select t2.id, t2.v, t3. from t2, t3 where t2.id = t3.id
+```
+
+```sql
+(1054, "Unknown column '3.id' in 'where clause'")
+```
+可以通过 dry run query 打印出查询语句来确认。
```
Comment (qiancai) [inline non-transactional-dml.md:341]:
```suggestion
当遇到此错误时，你可以通过 dry run query 打印出查询语句来确认。
```

File: non-transactional-dml.md [hunk 6]
```diff
@@ -327,6 +327,33 @@ batch-dml 是一种在 DML 语句执行期间将一个事务拆成多个事务
...
+
```
Comment (qiancai) [inline non-transactional-dml.md:342]: dry run query 具体是指什么

File: non-transactional-dml.md [hunk 7]
```diff
@@ -327,6 +327,33 @@ batch-dml 是一种在 DML 语句执行期间将一个事务拆成多个事务
...
+这种情况下可以尝试将 where 子句中涉及其它表的条件移动到 join 的 on 条件中，例如：
```
Comment (qiancai) [inline non-transactional-dml.md:343]:
```suggestion
要避免该错误，可以尝试将 `WHERE` 子句中涉及其它表的条件移动到 `JOIN` 的 `ON` 条件中，例如：
```

File: non-transactional-dml.md (hunk 3 above)
Comment (qiancai) [inline non-transactional-dml.md:332]: “划分列” 是什么意思

File: non-transactional-dml.md [hunk 8]
```diff
@@ -327,6 +327,33 @@ batch-dml 是一种在 DML 语句执行期间将一个事务拆成多个事务
...
+### 在多表 join 语句中，"Unknown column xxxx in 'where clause'" 错误
+
+这经常是拼接查询语句时，只查询划分列所在的表，但是 where 子句会直接拼接在这个查询语句中。如果 where 子句中的条件涉及到了其它表，就会出现这个错误。例如：
+
+```sql
+batch on test.t2.id limit 1 insert into t select t2.id, t2.v, t3. from t2, t3 where t2.id = t3.id
+```
+
+```sql
+(1054, "Unknown column '3.id' in 'where clause'")
```
Comment (qiancai) [inline non-transactional-dml.md:339]: 3.id 要改成 t3.id 吗

File: non-transactional-dml.md (hunk 6 above)
Comment (ekexium) [reply non-transactional-dml.md:342]: 这个文档里解释了划分列、dry run 等东西

File: non-transactional-dml.md (hunk 8 above)
Comment (ekexium) [reply non-transactional-dml.md:339]: 要😓

File: non-transactional-dml.md [hunk 9]
```diff
@@ -327,6 +327,33 @@ batch-dml 是一种在 DML 语句执行期间将一个事务拆成多个事务
...
+### 执行多表 join 语句后，遇到 `Unknown column xxxx in 'where clause` 错误
+
+这经常是拼接查询语句时，只查询划分列所在的表，但是 where 子句会直接拼接在这个查询语句中。如果 where 子句中的条件涉及到了其它表，就会出现这个错误。例如：
+
+```sql
+batch on test.t2.id LIMIT 1 INSERT INTO t SELECT t2.id, t2.v, t3. FROM t2, t3 WHERE t2.id = t3.id
+```
+
+```sql
+(1054, "Unknown column '3.id' in 'where clause'")
```
Comment (qiancai) [inline non-transactional-dml.md:339]:
```suggestion
(1054, "Unknown column 't3.id' in 'where clause'")
```

File: non-transactional-dml.md [hunk 10]
```diff
@@ -327,6 +327,34 @@ batch-dml 是一种在 DML 语句执行期间将一个事务拆成多个事务
 
 ## 常见问题
 
+### 执行多表 join 语句后，遇到 `Unknown column xxxx in 'where clause` 错误
```
Comment (qiancai) [inline non-transactional-dml.md:330]:
```suggestion
### 执行多表 join 语句后，遇到 `Unknown column xxx in 'where clause'` 错误
```

File: non-transactional-dml.md [hunk 11]
```diff
@@ -327,6 +327,34 @@ batch-dml 是一种在 DML 语句执行期间将一个事务拆成多个事务
...
+
+这经常是拼接查询语句时，只查询划分列所在的表，但是 where 子句会直接拼接在这个查询语句中。如果 where 子句中的条件涉及到了其它表，就会出现这个错误。例如：
```
Comment (qiancai) [inline non-transactional-dml.md:332]:
```suggestion
当进行多表 jon 时，如果将 `WHERE` 子句直接拼接在查询语句中且 `WHERE` 子句中的条件涉及到了 `SELECT` 子句中未指定的列，就会出现这个错误。例如：
```

Comment (ekexium) [reply non-transactional-dml.md:332]: 不太准确。一定会拼接 where 子句。触发条件是 where 子句里出现了拆分列所在的表以外的表。

File: non-transactional-dml.md [hunk 12]
```diff
@@ -327,6 +327,34 @@ batch-dml 是一种在 DML 语句执行期间将一个事务拆成多个事务
 
 ## 常见问题
 
+### 执行多表 join 语句后，遇到 `Unknown column xxx in 'where clause'` 错误
+
+这经常是拼接查询语句时，只查询划分列所在的表，但是 where 子句会直接拼接在这个查询语句中。如果 where 子句中的条件涉及到了其它表，就会出现这个错误。例如：
+
+```sql
+batch on test.t2.id LIMIT 1 INSERT INTO t SELECT t2.id, t2.v, t3. FROM t2, t3 WHERE t2.id = t3.id
```
Comment (qiancai) [inline non-transactional-dml.md:335]:
```suggestion
BATCH ON test.t2.id LIMIT 1 INSERT INTO t SELECT t2.id, t2.v, t3. FROM t2, t3 WHERE t2.id = t3.id
```

File: non-transactional-dml.md [hunk 13]
```diff
@@ -327,6 +327,34 @@ batch-dml 是一种在 DML 语句执行期间将一个事务拆成多个事务
...
+```
+
+```sql
+(1054, "Unknown column 't3.id' in 'where clause'")
+```
+
+当遇到此错误时，你可以通过 dry run query 打印出查询语句来确认。
```
Comment (qiancai) [inline non-transactional-dml.md:342]:
```suggestion
当遇到此错误时，你可以通过 `DRY RUN QUERY` 打印出查询语句来确认。例如：

```sql
(1054, "Unknown column 't3.id' in 'where clause'")
```
BATCH ON test.t2.id LIMIT 1 DRY RUN QUERY INSERT INTO t SELECT t2.id, t2.v, t3. FROM t2, t3 WHERE t2.id = t3.id
```

File: non-transactional-dml.md [hunk 14]
```diff
@@ -327,6 +327,34 @@ batch-dml 是一种在 DML 语句执行期间将一个事务拆成多个事务
...
+
+要避免该错误，可以尝试将 `WHERE` 子句中涉及其它表的条件移动到 `JOIN` 的 `ON` 条件中，例如：
+
+```sql
+batch on test.t2.id LIMIT 1 INSERT INTO t SELECT t2.id, t2.v, t3. FROM t2 JOIN t3 ON t2.id=t3.id
```
Comment (qiancai) [inline non-transactional-dml.md:347]:
```suggestion
BATCH ON test.t2.id LIMIT 1 INSERT INTO t SELECT t2.id, t2.v, t3. FROM t2 JOIN t3 ON t2.id=t3.id
```

File: non-transactional-dml.md [hunk 15]
```diff
@@ -327,6 +327,38 @@ batch-dml 是一种在 DML 语句执行期间将一个事务拆成多个事务
...
+(1054, "Unknown column 't3.id' in 'where clause'")
+```
+
+当遇到此错误时，你可以通过 `DRY RUN QUERY` 打印出查询语句来确认。例如：
+
+```sql
+BATCH ON test.t2.id LIMIT 1 DRY RUN QUERY INSERT INTO t SELECT t2.id, t2.v, t3. FROM t2, t3 WHERE t2.id = t3.id
+```
+
+要避免该错误，可以尝试将 `WHERE` 子句中涉及其它表的条件移动到 `JOIN` 的 `ON` 条件中，例如：
```
Comment (qiancai) [inline non-transactional-dml.md:348]:
```suggestion
要避免该错误，可以尝试将 `WHERE` 子句中涉及其它表的条件移动到 `JOIN` 的 `ON` 条件中。例如：
```

**Review by you06 — APPROVED** [review]

**Review by qiancai — APPROVED** [review]

**Pull request merged**